presto-main/src/main/java/com/facebook/presto/cost/FilterStatsCalculator.java

```diff
@@ -219,7 +219,8 @@ private PlanNodeStatsEstimate symbolToSymbolEquality(Symbol left, Symbol right)
             SymbolStatsEstimate leftStats = input.getSymbolStatistics().get(left);
             SymbolStatsEstimate rightStats = input.getSymbolStatistics().get(right);
 
-            if (isNaN(leftStats.getDistinctValuesCount()) || isNaN(rightStats.getDistinctValuesCount())) {
+            if (leftStats == null || rightStats == null ||
```

Comment:
If you use `input.getSymbolStatistics(left)` instead `input.getSymbolStatistics().get(left)` you will not need to check for null.